### PR TITLE
adding api DecrementAndWake to intelocked_hl

### PR DESCRIPTION
**Pull request by haridamineni**

### CODE REVIEW

File: src/interlocked_hl.c [hunk 1]
```diff
@@ -305,3 +305,27 @@ IMPLEMENT_MOCKABLE_FUNCTION(, INTERLOCKED_HL_RESULT, InterlockedHL_SetAndWakeAll
     return result;
 
 }
+
+IMPLEMENT_MOCKABLE_FUNCTION(, INTERLOCKED_HL_RESULT, InterlockedHL_DecrementAndWake, int32_t volatile_atomic*, address)
```
Comment (dcristoloveanu) [inline src/interlocked_hl.c:309]: >InterlockedHL_DecrementAndWake [](http://example.com/codeflow?start=53&length=30)

Missing unit (and int) tests.
Shall wait for that.

**Review by dcristoloveanu — CHANGES_REQUESTED** [review]
:clock1:

File: src/interlocked_hl.c (hunk 1 above)
Comment (haridamineni) [reply src/interlocked_hl.c:309]: the tests will come in another PR. let me know if they need to be included in the same PR.

Comment (dcristoloveanu) [reply src/interlocked_hl.c:309]: Yes, at least the UT we want in the same PR, please. I see there are no int tests for Interlocked_HL. It is Ok to add a new int test project as a separate PR, but at least the UT should be done together with the implementation. That's our process! :-)

Comment (haridamineni) [reply src/interlocked_hl.c:309]: unit test changes are implemented

File: tests/interlocked_hl_ut/interlocked_hl_ut.c [hunk 2]
```diff
@@ -867,4 +873,56 @@ TEST_FUNCTION(InterlockedHL_SetAndWakeAll_succeeds)
...
+    INTERLOCKED_HL_RESULT result;
+
+    // act
+    result = InterlockedHL_DecrementAndWake(NULL);
+
+    // assert
+    ASSERT_ARE_EQUAL(INTERLOCKED_HL_RESULT, INTERLOCKED_HL_ERROR, result);
+}
+
+TEST_FUNCTION(InterlockedHL_DecrementAndWake_with_Valid_Inputs_success)
```
Comment (dcristoloveanu) [inline tests/interlocked_hl_ut/interlocked_hl_ut.c:891]: >InterlockedHL_DecrementAndWake_with_Valid_Inputs_success [](http://example.com/codeflow?start=14&length=56)

Seems like a missing Tests_SRS for this test

Comment (dcristoloveanu) [reply tests/interlocked_hl_ut/interlocked_hl_ut.c:891]: Oh I see, you put them in ln. 912.

Comment (haridamineni) [reply tests/interlocked_hl_ut/interlocked_hl_ut.c:894]: modified and put the SRS in the function header

File: tests/interlocked_hl_ut/interlocked_hl_ut.c [hunk 3]
```diff
@@ -867,4 +873,56 @@ TEST_FUNCTION(InterlockedHL_SetAndWakeAll_succeeds)
...
+        /*Input*/              /*Output*/
+        { INT32_MAX,            INT32_MAX - 1},
+        { INT32_MAX - 1,        INT32_MAX - 2},
+        { INT32_MIN,            INT32_MAX},
+        { INT32_MIN + 1,        INT32_MIN},
+        { -1,                   -2 },
+        { 0,                    -1} ,
+        { 1,                    0},
+        { 1024,                 1023 },
+        { -1024,                -1025 },
```
Comment (dcristoloveanu) [inline tests/interlocked_hl_ut/interlocked_hl_ut.c:908]: Love this table.

**Review by dcristoloveanu — APPROVED** [review]
:shipit:

**Pull request merged**